File: Python3/1813. Sentence Similarity III.py
class Solution:
    def areSentencesSimilar(self, sentence1: str, sentence2: str) -> bool:
        words1 = sentence1.split(" ")
        words2 = sentence2.split(" ")

        while words1 and words2 and words1[0] == words2[0]:
            words1.pop(0)
            words2.pop(0)

        while words1 and words2 and words1[-1] == words2[-1]:
            words1.pop(-1)
            words2.pop(-1)

        return not words1 or not words2

File: Java/1813. Sentence Similarity III.java
class Solution {
    public boolean areSentencesSimilar(String sentence1, String sentence2) {
        Deque<String> words1 = new ArrayDeque<>(Arrays.asList(sentence1.split(" ")));
        Deque<String> words2 = new ArrayDeque<>(Arrays.asList(sentence2.split(" ")));

        while (!words1.isEmpty() && !words2.isEmpty() &&
                words1.peek().equals(words2.peek())) {
                    words1.poll();
                    words2.poll();
                }

        while (!words1.isEmpty() && !words2.isEmpty() &&
                words1.peekLast().equals(words2.peekLast())) {
                    words1.pollLast();
                    words2.pollLast();
                }

        return words1.isEmpty() || words2.isEmpty();
    }
}